src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -730,14 +730,19 @@ void ApplyTransparentBarBackground()
 			}
 			else
 			{
-				NavigationBar.SetBackgroundImage(new UIImage(), UIBarMetrics.Default);
-				NavigationBar.ShadowImage = new UIImage();
-				NavigationBar.BackgroundColor = UIColor.Clear;
-				NavigationBar.TintColor = UIColor.Clear;
-				NavigationBar.BarTintColor = UIColor.Clear;
+				SetTransparentNavigationBar (NavigationBar);
 			}		
 		}
 
+		internal static void SetTransparentNavigationBar (UINavigationBar navigationBar)
```

Comment:
can we move this to `NavigationPageExtensions`? 

Also can we move all the code related to setting the navbar into the Extentions method?

I think the iOS15+ code could also live inside the same method and then inside that method it does an operating system check?

Reply:
I moved the non-iOS15 code to the NavigationPageExtensions. I'm not sure if moving the ios 15 code there works too well. The iOS 15 code uses the UINavigationBarAppearance and the older code uses UINavigationBar. I could do an overload but since there are other things that get set in the iOS 15 check and the code for the iOS 15 stuff is only a one-liner, I'm not sure if that'd be too helpful. What do you think?